api/docs/mapstruct.md

```diff
@@ -0,0 +1,608 @@
+    }
+  }
+}
+```
+
+I thought I'd be clever and just pull in the enum type via a `uses` directive at the top of the mapper:
+```java
+@Mapper(uses = Department.class)
+```
+This did exactly the wrong thing. It generated the class without errors, but then failed to compile:
+```java
+public class EmployeeMapperImpl implements EmployeeMapper {
+
+    private final Department department = new Department();
+
+    @Override
+    public EmployeeModel toModel(EmployeeDbEntity employeeDbEntity) {
+        if ( employeeDbEntity == null ) {
+            return null;
+        }
+
+        EmployeeModel employeeModel = department.clone();
+
+        employeeModel.setName( toNickname( employeeDbEntity ) );
+        employeeModel.setAddress( employeeDbEntity.getAddress() );
+
+        return employeeModel;
+    }
+}
+```
+
+New errors:
+```
+> Task :compileJava 
+/Users/jaycarlton/repos/workbench/api/src/main/java/org/pmiops/workbench/utils/mappers/examples/EmployeeMapper.java:10: warning: Unmapped target properties: "department, salary".
+  EmployeeModel toModel(EmployeeDbEntity employeeDbEntity);
+                ^
+/Users/jaycarlton/repos/workbench/api/build/generated/sources/annotationProcessor/java/main/org/pmiops/workbench/utils/mappers/examples/EmployeeMapperImpl.java:12: error: enum types may not be instantiated
+    private final Department department = new Department();
+                                          ^
+/Users/jaycarlton/repos/workbench/api/build/generated/sources/annotationProcessor/java/main/org/pmiops/workbench/utils/mappers/examples/EmployeeMapperImpl.java:20: error: clone() has protected access in Enum
+        EmployeeModel employeeModel = department.clone();
+                                                ^
+/Users/jaycarlton/repos/workbench/api/build/generated/sources/annotationProcessor/java/main/org/pmiops/workbench/utils/mappers/examples/EmployeeMapperImpl.java:20: error: incompatible types: Object cannot be converted to EmployeeModel
+        EmployeeModel employeeModel = department.clone();
+                                                      ^
+```
+
+Fine, we can do it the less slick but more reliable way:
+
+```java
+@Mapper(uses = SampleEnumMapper.class)
+public interface EmployeeMapper {
+
+  @Mapping(source = "employeeDbEntity", target = "name")
+  @Mapping(source = "employeeDbEntity.departmentCode", target = "department")
+  EmployeeModel toModel(EmployeeDbEntity employeeDbEntity);
+
+  default String toNickname(EmployeeDbEntity employeeDbEntity) {
+    return String.format("%s %s", employeeDbEntity.getFirstName(), employeeDbEntity.getLastName());
+  }
+}
+```
+
+where the new `SampleEnumMapper` looks like:
+```java
+@Mapper
+public interface SampleEnumMapper {
+  default Department toDepartment(int departmentCode) {
+    return Department.fromDepartmentCode(departmentCode);
+  }
+}
+```
+
+I'm sure there's a way to do this without creating another mapper class. Anyway, this gives us 
```

Comment:
I mean, I am sure you could do it the same way you did name in the same class, but that is less reusable.

Reply:
A Mapper can use another Mapper, so as long as you avoid cycles it's still reusable.